### ISSUE TITLE
Syntax error at or near "CREDENTIALS" 

### ISSUE THREAD

**Issue by Tharaka99** [opt-tech/redshift-fake-driver#80]
I tried to use `redshift-fake-driver` in a Micronaut Java project.
I correctly setup the driver to `postgresqlredshift`  and the class name to `FakePostgresqlDriver`.
```
package org.example.util;

import javax.sql.DataSource;

import io.micronaut.context.annotation.Factory;
import jakarta.inject.Singleton;
import org.apache.commons.dbcp.BasicDataSource;
@Factory
public class RedshiftDataSourceFactory {

    @Singleton
    public DataSource createDataSource() {
        BasicDataSource dataSource = new BasicDataSource();
        dataSource.setUrl("jdbc:postgresqlredshift://localhost:5432/postgres");
        dataSource.setUsername("postgres");
        dataSource.setPassword("password");
        dataSource.setDriverClassName("jp.ne.opt.redshiftfake.postgres.FakePostgresqlDriver");
        return dataSource;
    }
}
```
But when I am running it I can see it's passing the query as it is to Postgress without any translation. So I am getting this Syntax error.
I can see it's going through `FakeStatement` , so I think there is no issues with the setup.
```
org.postgresql.util.PSQLException: ERROR: syntax error at or near "CREDENTIALS"
  Position: 111
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2510)
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2245)
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:311)
	at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:447)
	at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:368)
	at org.postgresql.jdbc.PgStatement.executeWithFlags(PgStatement.java:309)
	at org.postgresql.jdbc.PgStatement.executeCachedSql(PgStatement.java:295)
	at org.postgresql.jdbc.PgStatement.executeWithFlags(PgStatement.java:272)
	at org.postgresql.jdbc.PgStatement.executeUpdate(PgStatement.java:246)
	at jp.ne.opt.redshiftfake.FakeStatement$$anonfun$executeUpdate$1.apply$mcI$sp(FakeStatement.scala:101)
	at jp.ne.opt.redshiftfake.FakeStatement$$anonfun$executeUpdate$1.apply(FakeStatement.scala:101)
	at jp.ne.opt.redshiftfake.FakeStatement$$anonfun$executeUpdate$1.apply(FakeStatement.scala:101)
	at scala.Option.getOrElse(Option.scala:121)
	at jp.ne.opt.redshiftfake.FakeStatement.switchExecute(FakeStatement.scala:126)
	at jp.ne.opt.redshiftfake.FakeStatement.executeUpdate(FakeStatement.scala:101)
	at org.apache.commons.dbcp.DelegatingStatement.executeUpdate(DelegatingStatement.java:228)
	at org.apache.commons.dbcp.DelegatingStatement.executeUpdate(DelegatingStatement.java:228)
	at org.example.services.DatabaseService.copyData(DatabaseService.java:104)
```
My copy command is as below:
```
COPY product1_customer_entitlement FROM 
's3://sample-bucket/root-path/product1/customer_entitlement/data.csv' 
CREDENTIALS 'aws_access_key_id=accessKey;aws_secret_access_key=secretToken'  
region 'us-east-1' 
FILLRECORD;
```
I am running S3 locally using LocalStack, so I used this system properties while running:
```
-Dfake.awsRegion=us-east-1
-Dfake.awsS3Endpoint=http://localhost:4566/
```

If I change the copy command to support Postgress it goes a bit forward and then complains that the file in S3 is not accessible.

**Comment by Tharaka99** [comment]
I debugged and found the issue.  Since I am using a fake S3 (I used Local Stack), I have to use the same endpoint I have configured in the `-Dfake.awsS3Endpoint` system property in the S3 path as well.
Even though it's a bit wired with Local Stack, I have to change the S3 path to `http://localhost:4566/sample-bucket/root-path/product1/customer_entitlement/data.csv` in my example query.

So this is how the corrected query should looks like.
```
COPY product1_customer_entitlement FROM 
'http://localhost:4566/sample-bucket/root-path/product1/customer_entitlement/data.csv' 
CREDENTIALS 'aws_access_key_id=accessKey;aws_secret_access_key=secretToken'  
region 'us-east-1' 
FILLRECORD;
```

**Comment by Tharaka99** [comment]
I debugged and found the issue.  Since I am using a fake S3 (I used Local Stack), I have to use the same endpoint I have configured in the `-Dfake.awsS3Endpoint` system property in the S3 path as well.
So in my example query, I have to change the S3 path to `http://localhost:4566/sample-bucket/root-path/product1/customer_entitlement/data.csv` 

So this is how the corrected query should looks like.
```
COPY product1_customer_entitlement FROM 
'http://localhost:4566/sample-bucket/root-path/product1/customer_entitlement/data.csv' 
CREDENTIALS 'aws_access_key_id=accessKey;aws_secret_access_key=secretToken'  
region 'us-east-1' 
FILLRECORD;
```

**Comment by Tharaka99** [comment]
I debugged and found the issue.  Since I am using a fake S3 (I used Local Stack), I have to use the same endpoint I have configured in the `-Dfake.awsS3Endpoint` system property in the S3 path as well.
So in my example query, I have to change the S3 path to `http://localhost:4566/sample-bucket/root-path/product1/customer_entitlement/data.csv` 

So this is how the corrected query should looks like.
```
COPY product1_customer_entitlement FROM 
'http://localhost:4566/sample-bucket/root-path/product1/customer_entitlement/data.csv' 
CREDENTIALS 'aws_access_key_id=accessKey;aws_secret_access_key=secretToken'  
region 'us-east-1' 
FILLRECORD;
```
I am closing this issue as it's not an issue
